enhancers_pod.go

```diff
@@ -81,14 +83,32 @@ func runPodEnhancer(ctx context.Context, podMeta *v1.ObjectReference, cachedObje
 		sentryEvent.Fingerprint = []string{message}
 	}
 
-	if len(pod.OwnerReferences) > 0 {
-		// If the pod is controlled by something (e.g. a replicaset), group all issues
-		// for all controlled pod together.
-		owner := pod.OwnerReferences[0]
-		sentryEvent.Fingerprint = append(sentryEvent.Fingerprint, owner.Kind, owner.Name)
-	} else {
+	// Using finger print to group events together
+	// The pod is not owned by a higher resource
+	if len(pod.OwnerReferences) == 0 {
 		// Standalone pod => most probably it has a unique name
 		sentryEvent.Fingerprint = append(sentryEvent.Fingerprint, podName)
+
+		// The pod is owned by a higher resource
+	} else {
+
+		// error set to not nil of cronJob does not exist
+		var err error = nil
+
+		// only check for cronJob if the immediate pod owner
+		// is a job
+		if pod.OwnerReferences[0].Kind == "Job" {
+			// check if pod is part of cronJob (as grandchild workload resource)
+			err = runCronsDataHandler(ctx, scope, pod, sentryEvent)
+		}
+
+		// The job is not owned by a cronJob
+		// either because the pod is not owned by a job
+		// or the crons enhancer did not find a cronJob
+		if pod.OwnerReferences[0].Kind != "Job" || err != nil {
```

Comment:
To avoid this additional `!= "Job"` check, let's maybe return another value from `runCronsDataHandler`, e.g. a bool to indicate "the handler was run or not".

Reply:
The handler `runCronsDataHandler` returns an error is the pod is not associated with a cronJob. In the case where ownerReferences is not an empty slice, what if we always call `runCronsDataHandler` and check if the returned error is `nil`.

Reply:
Semantics is a bit messy in this case: it's not really an error that the given pod doesn't belong to a job/cronjob.
I'm suggesting to return `(handled bool, err error)` from `runCronsDataHandler`
I'm also thinking about further extensions. Let's say that we later want to improve our handling for cases when the pod is owned by a Deployment (more accurately, by a ReplicaSet, which is then owned by a Deployment). We probably don't want to unconditionally execute `runXxxDataHandler` for supported event type, right? 
I think it'll be more readable/reasonable to have a switch based on the owner type, and then attempt to run those handlers.